docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -0,0 +1,133 @@
+# Frontend Templates
+
+- **Status**: Draft (rev 3)
+- **Owner:** Altynbek Orumbayev
+- **Deciders**: TBD
+- **Date created**: 2023-06-06
+- **Date decided:** TBD
+- **Date updated**: 2023-06-08
+
+## Context
+
+AlgoKit v2 aims provide an end-to-end development and deployment experience that includes support for the end-to-end smart contract and dApp development lifecycle. With the release of the typed clients feature - developers are now able to reduce the time it takes to fully integrate the interactions between the contract and the frontend components powering the end-to-end dapp user experience. Hence, as a logical continuation, the following Architecture Decision Record aims to expand on the current capabilities of AlgoKit to support the `frontend` templates, an [AlgoKit Principles](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/algokit.md#guiding-principles) compliant approach towards simplifying integrations of smart contracts with the dApp frontend components.
+
+## Requirements
+
+- **Modularity**: The dApp templating feature should follow guiding AlgoKit principles and expand on approaches already utilized in existing smart contract templates feature. This implies that giving developers flexibility to mix and match different `smart contract` templates with `frontend` templates should serve as key consideration.
+- **Maintainability**: The dApp templating feature should be easy to maintain and extend. This implies that the feature should be implemented in a way that allows for easy addition of new templates and/or modification of existing ones as the complexity and variety of templates scale.
+- **Seamless onramp**: The dApp templating feature should provide a seamless onramp for developers to get started with dApp development. This implies that the feature should provide a simple and intuitive way to get started with dApp development and deployment. Providing developers a choice on whether they want more flexibility or rely on default recommended practices.
+
+## Principles
+
+- [AlgoKit Guiding Principles](../../docs/algokit.md#Guiding-Principles) - all of the aforementioned requirements should be met in a way that is consistent with the guiding principles of AlgoKit or attempt to find a balanced trade of between the principles that satisfies the requirements.
+
+## Options
+
+Expanding AlgoKit templating capabilities is a non-trivial task that requires careful consideration of the trade-offs between the different approaches. The following section outlines the different options that were considered and the trade-offs associated with each of them.
+
+> Refer to git history to see old revisions of the document. After further discussions with the team we are now aligned on the separate repositories approach. Hence the sections below are updated to reflect the new approach.
```

Comment:
Rather than doing this if we had other ptions we considered we should include them in the decision record so we can articulate the pros/cons and why we didn't explore them.